full-stack/orders-filters-and-listing.md

```diff
@@ -0,0 +1,128 @@
+# SteelEye full-stack developer - Orders filters and listing problem
+
+The question below is meant to give candidates a sense of the problems we tackle at SteelEye. Please submit solutions in the form of a readme + working code. The problems should take around three hours to complete (2 hrs for the APIs and 1 for the Frontend).
+
+Write an API endpoint that returns a paginated filtered list of Order records adn create a frontend for the same.
+
+## Requirements:
+* API - Python (use one of the Python framework: FastAPI, Flask or Django)
+* Database - Elasticsearch
+* Frontend - React.
+
+## Order record schema:
+```
+{
+  &id: { type: String, unique: True, required: True },
+  &model: { type: String, required: True }, // For order records value is 'Order'
+  orderId: { type: String, unique: True, required: True },
+  assetClass: { type: String, required: True, enum: [ 'Equity', 'Debt Instruments', 'Foreign Exchange' ] },
+  instrumentName: { type: String, required: True },
+  instrumentId: { type: String, required: True },
+
+  purchaseType: { type: String, required: True, enum: [ 'BUY', 'SELL' ] },
+  countparty: {
+    name: { type: String, required: True },
+  },
+
+  dataSource: {
+    name: { type: String, required: True },
+  },
+
+  tradeData: {
+    quantity: { type: String, required: True },
+    price: { type: Number, required: True },
+    trader: { type: String, required: True },
+  },
+
+  timestamps: {
+    orderSubmitted: { type: ISODateString, required: True },
+  }
+}
```

Comment:
Technically this isn't even an Elasticsearch "schema" (mapping as it's called in ES terminology). This is our own internal description of a schema model and we have our own internal tools which convert this to an Elasticsearch mapping.

Are you expect the candidate to take this JSON and create their own ES Mapping? If so, I would mention that explicitly. If _not_, then I would provide the mapping for them.

Reply:
> I was more referring to the fact that you can request the candidate to use certain frameworks rather than leaving it wide-open (e.g. pick _any_ Python or Front-end framework).
> 
> However, are you expecting to find a full-stack candidate with ES experience as well?
> 
> @bacon1664 makes a valid point, what is the expectation for the candidate when it comes to Elasticsearch? Or maybe more-broadly, what is the expectation for the full-stack developer role? Should they know how to spin up their own Elasticsearch instance using Docker, create an index with a mapping based on the JSON that you provided?

@sdsalamida @bacon1664 Yes, we want a full-stack candidate with Python + ES experience a and a bit of React as well.

I've created an ES instance where I have uploaded dummy data based on the given schema. User just need to run the `_search` query to get the results.